DelvUI/Interface/HudHelper.cs

```diff
@@ -0,0 +1,246 @@
+
+        private bool _previousCombatState = true;
+        public bool UserInterfaceWasHidden = false;
+
+        public HudHelper()
+        {
+            Config.onValueChanged += ConfigValueChanged;
+        }
+
+        ~HudHelper()
+        {
+            Config.onValueChanged -= ConfigValueChanged;
+        }
+
+        #region Event Handlers
+
+        public void ConfigValueChanged(object sender, OnChangeBaseArgs e)
+        {
+            if (e.KeyName == "HideDefaultCastbar")
+            {
+                ConfigureDefaultCastBar();
+            }
+            else if (e.KeyName == "HideDefaultJobGauges")
+            {
+                ConfigureDefaultJobGauge();
+            }
+            else if (e.KeyName == "EnableCombatActionBars" && e is OnChangeEventArgs<bool> boolEvent)
+            {
+                Config.CombatActionBars.ForEach(name => ToggleActionbar(name, boolEvent.Value));
+            }
+            else if (e.KeyName == "CombatActionBars" && e is OnChangeEventArgs<string> listEvent)
+            {
+                switch (listEvent.ChangeType)
+                {
+                    case ChangeType.ListAdd:
+                        ToggleActionbar(listEvent.Value, !IsInCombat());
+                        break;
+                    case ChangeType.ListRemove:
+                        ToggleActionbar(listEvent.Value, false);
+                        break;
+                }
+            }
+
+        }
+
+        #endregion
+
+
+        #region Action Bars
+        public void ConfigureCombatActionBars()
+        {
+            if (!Config.EnableCombatActionBars)
+            {
+                return;
+            }
+
+            var currentCombatState = IsInCombat();
+            if (_previousCombatState != currentCombatState && Config.CombatActionBars.Count > 0 || UserInterfaceWasHidden)
+            {
+                Config.CombatActionBars.ForEach(name => ToggleActionbar(name, !currentCombatState));
+                _previousCombatState = currentCombatState;
+            }
+        }
+
+        #endregion
+
+        public bool IsElementHidden(HudElement element = null)
+        {
+            if (!ConfigurationManager.GetInstance().LockHUD)
+            {
+                return ConfigurationManager.GetInstance().LockHUD;
+            }
+
+            bool isHidden = Config.HideOutsideOfCombat && !IsInCombat();
+
+            if (element != null)
+            {
+                if (element.ID == HudManager.playerCastbarID)
+                {
+                    return false;
+                }
+
+                if (element.ID == HudManager.playerUnitFrameID)
+                {
+                    PlayerCharacter player = Plugin.ClientState.LocalPlayer;
+                    Debug.Assert(player != null, "HudHelper.LocalPlayer is NULL.");
+                    isHidden = isHidden && player.CurrentHp == player.MaxHp;
+                }
```

Comment:
I think you should either check directly for the type or add a json ignored property on the config objects.
The way you have it right now is not very flexible / scalable.

Reply:
for the castbar thats easy, but for the player UnitFrameHud there are few elements of that type, unless i miss something, or maybe i can check the UnitFrameHud.actor ?

Reply:
Theres a **PlayerUnitFrameConfig** class.